x-pack/plugins/ml/server/lib/ml_client/ml_client.ts

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchClient, IScopedClusterClient } from 'kibana/server';
+
+import { JobSavedObjectService, JobType } from '../../saved_objects';
+
+import {
+  Job,
+  JobStats,
+  Datafeed,
+  DatafeedStats,
+} from '../../../common/types/anomaly_detection_jobs';
+import { Calendar } from '../../../common/types/calendars';
+import { searchProvider } from './search';
+
+import { DataFrameAnalyticsConfig } from '../../../common/types/data_frame_analytics';
+import { MLJobNotFound } from './errors';
+
+type OrigMlClient = ElasticsearchClient['ml'];
+
+export interface MlClient extends OrigMlClient {
+  anomalySearch: ReturnType<typeof searchProvider>['anomalySearch'];
+}
+
+export function getMlClient(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+): MlClient {
+  const mlClient = client.asInternalUser.ml;
+
+  async function jobIdsCheck(jobType: JobType, p: any) {
+    const jobIds = getADJobIds(p);
```

Comment:
If this `jobIdsCheck` is also used for 'data-frame-analytics' type jobs - should `getADJobIds` be called something more generic?